pkg/apis/networking/validation/validation.go

```diff
@@ -206,27 +239,29 @@ func validateIngressTLS(spec *networking.IngressSpec, fldPath *field.Path) field
 }
 
 // ValidateIngressSpec tests if required fields in the IngressSpec are set.
-func ValidateIngressSpec(spec *networking.IngressSpec, fldPath *field.Path) field.ErrorList {
+func ValidateIngressSpec(spec *networking.IngressSpec, fldPath *field.Path, requestGV schema.GroupVersion) field.ErrorList {
 	allErrs := field.ErrorList{}
-	// TODO: Is a default backend mandatory?
-	if spec.Backend != nil {
-		allErrs = append(allErrs, validateIngressBackend(spec.Backend, fldPath.Child("backend"))...)
+	if spec.DefaultBackend != nil {
+		allErrs = append(allErrs, validateIngressBackend(spec.DefaultBackend, fldPath.Child("defaultBackend"))...)
 	} else if len(spec.Rules) == 0 {
-		allErrs = append(allErrs, field.Invalid(fldPath, spec.Rules, "either `backend` or `rules` must be specified"))
+		switch requestGV {
+		case networkingv1beta1.SchemeGroupVersion, extensionsv1beta1.SchemeGroupVersion:
+			allErrs = append(allErrs, field.Invalid(fldPath, spec.Rules, "either `backend` or `rules` must be specified"))
+		default:
+			allErrs = append(allErrs, field.Invalid(fldPath, spec.Rules, "either `defaultBackend` or `rules` must be specified"))
+		}
 	}
 	if len(spec.Rules) > 0 {
 		allErrs = append(allErrs, validateIngressRules(spec.Rules, fldPath.Child("rules"))...)
```

Comment:
is nil ok to be appended to the allErrs?

Reply:
Good question. Apparently both nil and an empty list are valid here. In this case when no errors are found, an empty error list is returned. 

Nil: https://play.golang.org/p/XyCrTYN-sLS
Empty List: https://play.golang.org/p/pKDiE0m1nTa